.github/workflows/flutter-branch.yml

```diff
@@ -68,6 +74,32 @@ jobs:
           working-directory: ${{ inputs.working-directory }}
       - run: flutter test --dart-define STAGE=mock
 
+  static-analysis:
+    runs-on: [self-hosted, Linux]
+    if: ${{ inputs.static-analysis }}
+    defaults:
+      run:
+        working-directory: ${{ inputs.working-directory }}
+    steps:
+      - uses: QuickBirdEng/actions/checkout-ssh@main
+        with:
+          ssh-private-key: ${{ secrets.CI_SSH_PRIVATE_KEY_FOR_GITHUB_PRIVATE_REPOS }}
+          git-lfs: ${{ inputs.git-lfs }}
+      - uses: QuickBirdEng/actions/setup-flutter@main
+        with:
+          working-directory: ${{ inputs.working-directory }}
+      - name: Download sonar-scanner
+        run: |
+          curl --create-dirs -sSLo $HOME/.sonar/sonar-scanner.zip https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-6.1.0.4477-linux-x64.zip
+          unzip -o $HOME/.sonar/sonar-scanner.zip -d $HOME/.sonar/
+      - name: Run sonar-scanner
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          SONAR_HOST_URL: ${{ vars.SONAR_HOST_URL }}
+        run: |
+          $HOME/.sonar/sonar-scanner-6.1.0.4477-linux-x64/bin/sonar-scanner \
```

Comment:
Would it make sense to extract the version + platform part here and above into a variable, so that it can be changed in one place?

Reply:
It makes sense, but access to the org vars is restricted so it puts it on you and few others to be able to change. So overall, I'd say that's your call

Reply:
Wait, never mind. My brain processed that as being about a GitHub variable, but it was still on my brain and it clicked what you meant. Actually we do use it in 2 workflows by the end, so I'll move it to the actions repo and just call it from there